accounts/abi/numbers.go

```diff
@@ -39,6 +39,7 @@ var (
 )
 
 // U256 converts a big Int into a 256bit EVM number.
+// This operation is destructive.
```

Comment:
I'm doubt that people will come read this. One idea would be to replace this function by two: `ToU256` and `AsU256` so that it is obvious from the function name that one of them converts and the other one recasts. I am not advocating for this to be changed in _this_ PR as you would need to get the approval of more people. Just keep it in mind for another refactoring.